ouroboros-consensus-shelley/test/Test/Consensus/Shelley/Ledger.hs

```diff
@@ -515,6 +514,22 @@ instance Arbitrary a => Arbitrary (StrictSeq a) where
 instance Arbitrary SL.PoolMetaData where
   arbitrary = (`SL.PoolMetaData` "bytestring") <$> arbitrary
 
+instance Arbitrary SL.Port where
+  arbitrary = fromIntegral @Word8 @SL.Port <$> arbitrary
+
+instance Arbitrary SL.IPv4 where
+  arbitrary = SL.mkIPv4 <$> arbitrary
+
+instance Arbitrary SL.IPv6 where
+  arbitrary = SL.mkIPv6 <$> arbitrary
```

Comment:
You'll have to limit the range, as the roundtrip tests are failing on CI: https://hydra.iohk.io/build/2444486/nixlog/13